lib/ui/login/login_screen.dart

```diff
@@ -108,102 +122,129 @@ class _LoginScreenState extends State<LoginScreen>
-                Padding(
-                  padding: const EdgeInsets.only(
-                    left: AppDimensions.spacing24,
-                    right: AppDimensions.spacing24,
-                    bottom: AppDimensions.spacing20,
-                  ),
-                  child: _configuredTextField(context, true),
+        ),
+        FadeTransition(
+          opacity: _overlayOpacityAnimationController,
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Padding(
+                padding: const EdgeInsets.only(
+                  left: AppDimensions.spacing24,
+                  right: AppDimensions.spacing24,
+                  bottom: AppDimensions.spacing20,
                 ),
-                Padding(
-                  padding: const EdgeInsets.only(
-                    left: AppDimensions.spacing24,
-                    right: AppDimensions.spacing24,
-                    bottom: AppDimensions.spacing20,
-                  ),
-                  child: _configuredTextField(context, false),
+                child: _configuredTextField(context, true),
+              ),
+              Padding(
+                padding: const EdgeInsets.only(
+                  left: AppDimensions.spacing24,
+                  right: AppDimensions.spacing24,
+                  bottom: AppDimensions.spacing20,
+                ),
+                child: _configuredTextField(context, false),
+              ),
+              Padding(
+                padding: const EdgeInsets.symmetric(
+                  horizontal: AppDimensions.spacing24,
                 ),
-                Padding(
-                  padding: const EdgeInsets.symmetric(
-                    horizontal: AppDimensions.spacing24,
-                  ),
-                  child: SizedBox(
-                    height: 56.0,
-                    width: double.infinity,
-                    child: _loginButton(context),
-                  ),
+                child: SizedBox(
+                  height: 56.0,
+                  width: double.infinity,
+                  child: _loginButton(context),
                 ),
-              ],
-            ),
+              ),
+            ],
           ),
-        ],
-      );
+        ),
+      ],
+    );
+  }
 
   @override
   Widget build(BuildContext context) {
+    final isLoginSucceed = ref.watch(_loginStatusProvider).value ?? false;
+    final errorMessage = ref.watch(_loginErrorStatusProvider).value ?? "";
     return Material(
-      child: _buildLoginScreen(context),
+      child: Scaffold(
+        body: ref.watch(loginViewModelProvider).when(
+              init: () =>
+                  _buildLoginScreen(context, isLoginSucceed, errorMessage),
+              loading: () =>
+                  _buildLoginScreen(context, isLoginSucceed, errorMessage),
+              loginSuccess: () =>
+                  _buildLoginScreen(context, isLoginSucceed, errorMessage),
```

Comment:
Hmmmm, is it necessary to build the login screen again if the login succeeded? 

I was expecting we could directly navigate to the home screen 💭

Reply:
@Tuubz Do you think we should navigate from the viewModel?

Reply:
@Shayokh144 From my understanding, we must use `listen` instead of `watch` (RiverPod)

![image](https://user-images.githubusercontent.com/18277915/232702341-f732e28d-1d4e-453d-ba0e-5b658042ba20.png)

Because instead of rebuilding a widget, we need to invoke a callback.

Please have a look at this example: https://github.com/chornerman/survey-flutter/blob/develop/lib/page/login/login_page.dart

This is very new for me too, please don't hesitate to reach out to me 🙏

Reply:
@Shayokh144 

Actually, I think your solution is acceptable too (since we're doing something similar for the error snack bar) 👍 

Please ignore my comment, I'll resolve it 🙇 

cc @luongvo